File: TickerQ.Utilities/Models/InternalFunctionContext.cs
using System;
using TickerQ.Utilities.Enums;

namespace TickerQ.Utilities.Models
{
    internal class InternalFunctionContext
    {
        public string FunctionName { get; set; }
        public Guid TickerId { get; set; }
        public TickerType Type { get; set; }
        public int Retries { get; set; }
        public int RetryCount { get; set; }
        public TickerStatus Status { get; set; }
        public long ElapsedTime { get; set; }
        public string ExceptionDetails { get; set; }
        public int[] RetryIntervals { get; set; }
    }
}

File: hub/sdks/node/src/models/InternalFunctionContext.ts
import { TickerType, TickerStatus, TickerTaskPriority, RunCondition } from '../enums';
import type { ResultEnvelope } from './ResultEnvelope';

export interface InternalFunctionContext {
    parametersToUpdate: string[];
    cachedPriority: TickerTaskPriority;
    cachedMaxConcurrency: number;
    functionName: string;
    tickerId: string;
    parentId: string | null;
    type: TickerType;
    retries: number;
    retryCount: number;
    status: TickerStatus;
    elapsedTime: number;
    exceptionDetails: string | null;
    executedAt: string;
    retryIntervals: number[];
    releaseLock: boolean;
    executionTime: string;
    runCondition: RunCondition;
    timeTickerChildren: InternalFunctionContext[];
    /** Included only in a final successful status update. */
    resultEnvelope?: ResultEnvelope;
}
